xpdacq/xpdacq.py

```diff
@@ -192,6 +192,58 @@ def _parse_calibration_file(config_file_name):
                 config_dict[option] = None
     return config_dict
 
+def new_prun(sample, scanplan, auto_dark = glbl.auto_dark, **kwargs):
+    '''on this 'sample' run this 'scanplan'
+        
+    Arguments:
+    sample - sample metadata object
+    scanplan - scanplan metadata object
+    auto_dark - optional. Type auto_dark = False to suppress the automatic collection of a dark image (default = True). Strongly recommend to leave as True unless problems are encountered
+    **kwargs - dictionary that will be passed through to the run-engine metadata
+    '''
+    scan = Scan(sample, scanplan)
+    scan.md.update({'sc_isprun':True})
+    if auto_dark:
+        auto_dark_md_dict = _auto_dark(scan)
```

Comment:
sounds great, this would be my vision too.....though the ``dark`` may be a
"special" kind of scan that doesn't follow this logic....I would be ok with
that.  For example, it is odd to have the autodark stuff inside of dark.
It will lead to some interesting circular logic that dark is called from
within itself.....

On Mon, Mar 21, 2016 at 10:50 AM, Timothy Liu <notifications@github.com>
wrote:

> In xpdacq/xpdacq.py
> <https://github.com/xpdAcq/xpdAcq/pull/156#discussion_r56834315>:
>
> > @@ -192,6 +192,58 @@ def _parse_calibration_file(config_file_name):
> >                  config_dict[option] = None
> >      return config_dict
> >
> > +def new_prun(sample, scanplan, auto_dark = glbl.auto_dark, **kwargs):
> > +    '''on this 'sample' run this 'scanplan'
> > +
> > +    Arguments:
> > +    sample - sample metadata object
> > +    scanplan - scanplan metadata object
> > +    auto_dark - optional. Type auto_dark = False to suppress the automatic collection of a dark image (default = True). Strongly recommend to leave as True unless problems are encountered
> > +    **kwargs - dictionary that will be passed through to the run-engine metadata
> > +    '''
> > +    scan = Scan(sample, scanplan)
> > +    scan.md.update({'sc_isprun':True})
> > +    if auto_dark:
> > +        auto_dark_md_dict = _auto_dark(scan)
>
> Yes, I am moving toward this direction. My goal of this PR is to have
> execute_scans() which includes automated logics (dark and load
> calibration parameters) and other necessary logics. dark, background...e.t.c
> will all call execute_scans(). The final code could look like
>
> def prun(sample,  scanplan, <options>):
>     scan = Scan(sample, scanplan)
>     scan.md.update({'is_prun':True})
>     _execute_scans(scan, <options>)
>     return
>
> def dark(sample,  scanplan, <options>):
>     scan = Scan(sample, scanplan)
>     scan.md.update({'is_dark':True})
>     _execute_scans(scan, <options>)
>     return
> ...
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/xpdAcq/xpdAcq/pull/156/files/b5cfa3989d4cbf78f9cbca5c529e98a22bbab4ac#r56834315>
>



-- 
Prof. Simon Billinge
Applied Physics & Applied Mathematics
Columbia University
500 West 120th Street
Room 200 Mudd, MC 4701
New York, NY 10027
Tel: (212)-854-2918 (o) 851-7428 (lab)

Condensed Matter Physics and Materials Science Dept.
Brookhaven National Laboratory
P.O. Box 5000
Upton, NY 11973-5000
(631)-344-5661

email: sb2896 at columbia dot edu
home: http:// <http://nirt.pa.msu.edu/>bgsite.apam.columbia.edu/